Recent GTK changes cause build error

I tried to build a recent commit (35878a3) from the master branch in Fedora, but I got the following error:
```
[ 86%] Building CXX object src/celestia/gtk/CMakeFiles/celestia-gtk.dir/dialog-solar.cpp.o
cd /builddir/build/BUILD/Celestia-35878a35f1982b381683cc46e26c4c61011b9d94/redhat-linux-build/src/celestia/gtk && /usr/bin/g++ -DCAIRO -DCELX -DCONFIG_DATA_DIR=\"/usr/share/celestia\" -DEIGEN_NO_DEBUG -DENABLE_NLS -DFMT_SHARED -DGIT_COMMIT=\"35878a3\" -DGTKGLEXT -DHIP_DATA_DIR=\"/usr/share/celestia\" -DLOCALEDIR=\"/usr/share/locale\" -DNO_DEBUG -DUSE_FFMPEG -DUSE_LIBAVIF -DUSE_MINIAUDIO -DVERSION=\"1.7.0\" -D_USE_MATH_DEFINES -I/builddir/build/BUILD/Celestia-35878a35f1982b381683cc46e26c4c61011b9d94/redhat-linux-build/src/celestia/gtk -I/builddir/build/BUILD/Celestia-35878a35f1982b381683cc46e26c4c61011b9d94/src/celestia/gtk -I/builddir/build/BUILD/Celestia-35878a35f1982b381683cc46e26c4c61011b9d94/src -I/builddir/build/BUILD/Celestia-35878a35f1982b381683cc46e26c4c61011b9d94/redhat-linux-build -I/usr/include/gtk-2.0 -I/usr/lib64/gtk-2.0/include -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/cairo -I/usr/include/harfbuzz -I/usr/include/glib-2.0 -I/usr/lib64/glib-2.0/include -I/usr/include/webp -I/usr/include/libmount -I/usr/include/blkid -I/usr/include/fribidi -I/usr/include/libxml2 -I/usr/include/pixman-1 -I/usr/include/libpng16 -I/usr/include/sysprof-6 -I/usr/include/gtkglext-1.0 -I/usr/lib64/gtkglext-1.0/include -I/builddir/build/BUILD/Celestia-35878a35f1982b381683cc46e26c4c61011b9d94/thirdparty/miniaudio -isystem /usr/include/freetype2 -isystem /usr/include/ffmpeg -isystem /usr/include/eigen3 -isystem /usr/include/luajit-2.1 -O2 -flto=auto -ffat-lto-objects -fexceptions -g -grecord-gcc-switches -pipe -Wall -Werror=format-security -Wp,-U_FORTIFY_SOURCE,-D_FORTIFY_SOURCE=3 -Wp,-D_GLIBCXX_ASSERTIONS -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -fstack-protector-strong -specs=/usr/lib/rpm/redhat/redhat-annobin-cc1  -m64 -march=x86-64 -mtune=generic -fasynchronous-unwind-tables -fstack-clash-protection -fcf-protection -fno-omit-frame-pointer -mno-omit-leaf-frame-pointer -DNDEBUG -std=gnu++17 -fPIE -fvisibility=hidden -fno-exceptions -Wno-deprecated-declarations -MD -MT src/celestia/gtk/CMakeFiles/celestia-gtk.dir/dialog-solar.cpp.o -MF CMakeFiles/celestia-gtk.dir/dialog-solar.cpp.o.d -o CMakeFiles/celestia-gtk.dir/dialog-solar.cpp.o -c /builddir/build/BUILD/Celestia-35878a35f1982b381683cc46e26c4c61011b9d94/src/celestia/gtk/dialog-solar.cpp
/builddir/build/BUILD/Celestia-35878a35f1982b381683cc46e26c4c61011b9d94/src/celestia/gtk/dialog-options.cpp: In function `void celestia::gtk::dialogViewOptions(AppData*)`:
/builddir/build/BUILD/Celestia-35878a35f1982b381683cc46e26c4c61011b9d94/src/celestia/gtk/dialog-options.cpp:217:44: error: cannot convert `GtkObject*` {aka `_GtkObject*`} to `GtkAdjustment*` {aka `_GtkAdjustment*`} in initialization
  217 |     GtkAdjustment *adj = gtk_adjustment_new(static_cast<gfloat>(logDistanceLimit * DistanceSliderRange),
      |                          ~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
      |                                            |
      |                                            GtkObject* {aka _GtkObject*}
  218 |                                             0.0, DistanceSliderRange, 1.0, 2.0, 0.0);
      |                                             ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
gmake[2]: *** [src/celestia/gtk/CMakeFiles/celestia-gtk.dir/build.make:135: src/celestia/gtk/CMakeFiles/celestia-gtk.dir/dialog-options.cpp.o] Error 1
```

I think that's related to #2106 
The full log is [here](https://download.copr.fedorainfracloud.org/results/mattia/Astronomy/fedora-rawhide-x86_64/07107427-celestia/builder-live.log.gz).
That's with gtk2 2.24.33

--- a/src/celestia/gtk/dialog-options.cpp
+++ b/src/celestia/gtk/dialog-options.cpp
@@ -214,8 +214,9 @@ dialogViewOptions(AppData* app)
     gtk_container_set_border_width(GTK_CONTAINER(hbox), CELSPACING);
 
     float logDistanceLimit = log(app->renderer->getDistanceLimit()) / log((float)MaxDistanceLimit);
-    GtkAdjustment *adj = gtk_adjustment_new(static_cast<gfloat>(logDistanceLimit * DistanceSliderRange),
-                                            0.0, DistanceSliderRange, 1.0, 2.0, 0.0);
+    // In GTK2, gtk_adjustment_new returns GtkObject* instead of GtkAdjustment*
+    auto adj = (GtkAdjustment*)gtk_adjustment_new(static_cast<gfloat>(logDistanceLimit * DistanceSliderRange),
+                                                  0.0, DistanceSliderRange, 1.0, 2.0, 0.0);
 
     /* Distance limit slider */
     GtkWidget* magLabel = gtk_label_new(nullptr);

